### ISSUE TITLE
Test dataset is shuffled but result is compared to labels of unshuffled data source

### ISSUE THREAD

**Issue by Voxelworld** [udacity/nd320-c1-emr-data-starter#4]
Hi @md14kpi 

There is a bug affecting the notebooks where the TF prediction is compared to the original labels.
In contrast to the [original source](https://www.tensorflow.org/tutorials/structured_data/feature_columns) of the `df_to_dataset` helper function the `shuffle` flag has been removed which results in shuffled test and validation sets. Nevertheless, the entries are compared row by row to the original labels when the models are evaluated.

Fix: reintroduce `shuffle` flag:
```
def df_to_dataset(df, predictor,  batch_size=32, shuffle=True):
    df = df.copy()
    labels = df.pop(predictor)
    ds = tf.data.Dataset.from_tensor_slices((dict(df), labels))
    if shuffle:
        ds = ds.shuffle(buffer_size=len(df))
    ds = ds.batch(batch_size)
    return ds
```

The following screenshot shows the fixed and original regression plot in [Lesson_4_screencast.ipynb](https://github.com/udacity/nd320-c1-emr-data-starter/blob/master/lesson-5-Building-Evaluating-and-Interpreting-Models-for-Bias-and-Uncertainty/Lesson_4_screencast.ipynb):

![bug](https://user-images.githubusercontent.com/49528432/90314092-5dd68780-df11-11ea-9c56-28a09aefd20f.png)
